cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -70,3 +70,47 @@ def test_minimize_ipopt_nojac_constraints_if_scipy():
     expected_res = np.array([1.001867, 0.99434067, 1.05070075, 1.17906312,
                              1.38103001])
     np.testing.assert_allclose(res.get("x"), expected_res)
+
+
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+def test_minimize_ipopt_jac_and_hessians_constraints_if_scipy():
+    """`minimize_ipopt` works with objective gradient and Hessian 
+       and constraint jacobians and Hessians."""
+    from scipy.optimize import rosen, rosen_der, rosen_hess
+    x0 = [1.3, 0.7, 0.8, 1.9, 1.2]
+    constr = {"fun": lambda x: rosen(x) - 1.0, "type": "ineq",
+              "jac": rosen_der, "hess": lambda x, v: rosen_hess(x) * v[0]}
+    res = cyipopt.minimize_ipopt(rosen, x0, jac=rosen_der, hess=rosen_hess,
+                                 constraints=constr)
+    assert isinstance(res, dict)
+    assert np.isclose(res.get("fun"), 1.0828541)
```

Comment:
Can you please explain why the expected result of this test is 1.082541 and not 1.0? The problem formulation seems to indicate that the result should in fact be 1.0...

Reply:
Good catch! Removing the Hessians, the result is 0.999999990217499. After testing a few more test problems, I noticed that there's definitely something wrong with the evaluation of the lagrangian hessian (corresponds to the methods `hessian` and `hessianstructure`) :/. I'll delve into it.

Reply:
Excellent, glad we caught :)

Reply:
Found it! :)  The formula for the number of non-zero elements in the hessian is wrong, see [here](https://github.com/mechmotum/cyipopt/blob/d1c8d70bcd3c2c186e611f710509d6cc144e123f/cyipopt/cython/ipopt_wrapper.pyx#L366). It should be `n*(n+1)/2` instead of `n*(n-1)/2`. This also (partially) fixes #125. 

Nonetheless, I still obtain an objective value of 1.08 with the 'mu_strategy'='adaptive' for both MWEs. However, both now give the correct solution with objective value 1.0000000552371922 with the default 'mu_strategy' ('monotone').
After having tested some other problems without any errors, I reckon that the above test problem is just a rare corner case in which the different 'mu_strategy' options yield different solutions. So I'd propose to change the above test problem and add some additional test problems (just to play safe). What do you think?